rllib/utils/actors.py

```diff
@@ -65,45 +69,111 @@ def count(self):
         return len(self._tasks)
 
 
+def create_colocated_actors(
+        actor_specs: Sequence[Tuple[Type, Any, Any, int]],
+        node: Optional[str] = "localhost",
+        max_attempts: int = 10,
+) -> Dict[Type, List[ActorHandle]]:
+    """Create co-located actors of any type(s) on any node.
+
+    Args:
+        actor_specs: Tuple/list with tuples consisting of: 1) The
+            (already @ray.remote) class(es) to construct, 2) c'tor args,
+            3) c'tor kwargs, and 4) the number of actors of that class with
+            given args/kwargs to construct.
+        node: The node to co-locate the actors on. By default ("localhost"),
+            place the actors on the node the caller of this function is
+            located on. Use None for indicating that any (resource fulfilling)
+            node in the clusted may be used.
+        max_attempts: The maximum number of co-location attempts to
+            perform before throwing an error.
+
+    Returns:
+        A dict mapping the created types to the list of n ActorHandles
+        created (and co-located) for that type.
+    """
+    if node == "localhost":
+        node = platform.host()
+
+    # Maps types to lists of already co-located actors.
+    ok = [[] for _ in range(len(actor_specs))]
+    attempt = 1
+    while attempt < max_attempts:
```

Comment:
if these actors don't fit on a same node the first time, why would they fit when we try the second time?
this is a case where we need PACK scheduling it seems.

Reply:
We do use "PACK" strategy by default, so this should be ok. But it's a good question: Could still be that ray places the actor on a different node (bundle), no? And then we have to try again. I would love to use a ray tool to force placing an actor on a given node, but I don't think this exists.

Reply:
yeah, I would love to show core folks this as an example use case when this PR is in.